huffman/src/detail/iterator_interface.hpp

```diff
@@ -75,7 +74,7 @@ struct iterator_interface
   constexpr auto
   operator[](typename I::difference_type n) const -> typename I::reference
   {
-    return *(*this + n);
+    return *(static_cast<const D&>(*this) + n);
```

Comment:
Why D and not I?

Reply:
Changed to `I` but both are the same in this case.